homework/index.js

```diff
@@ -30,18 +31,102 @@
+    });
+  }
+  const mainContainerForRepo = document.querySelector('.container');
+
+  function createDescription(selectedRepository) {
+    const descriptionContainer = createAndAppend('div', mainContainerForRepo, {
+      class: 'left-div',
+    });
+    const table = createAndAppend('table', descriptionContainer, { class: 'table' });
+    const tbody = createAndAppend('tbody', table);
+    const details = [
+      { title: 'Repository', value: selectedRepository.name },
+      { title: 'Description', value: selectedRepository.description },
+      { title: 'Forks', value: selectedRepository.forks },
+      { title: 'Updated', value: selectedRepository.updated_at },
+    ];
+    details.forEach(detail => {
+      const tr = createAndAppend('tr', tbody);
+      createAndAppend('td', tr, {
+        text: detail.title,
+        class: `label `,
+      });
+      createAndAppend('td', tr, { text: detail.value, class: 'repository-data' });
+    });
+  }
+
+  function createContributors(selectedRepository, root) {
+    const contributorsContainer = createAndAppend('div', mainContainerForRepo, {
+      class: 'right-div',
+    });
+    createAndAppend('h3', contributorsContainer, {
+      text: 'Contributors',
+      class: 'contributors-header',
+    });
+    const ul = createAndAppend('ul', contributorsContainer, { class: 'contributor-list' });
+    fetchJSON(selectedRepository.contributors_url, (err, ShowContributors) => {
+      if (err) {
+        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+      } else {
+        ShowContributors.forEach(contributor => {
+          const li = createAndAppend('li', ul, {
+            class: 'contributor-item',
+          });
+          const alink = createAndAppend('a', li);
+          const dataDiv = createAndAppend('div', alink, { class: 'contributor' });
+
+          createAndAppend('img', dataDiv, {
+            class: 'contributor-item',
+            src: contributor.avatar_url,
+            height: 52,
+          });
+          const cotributorData = createAndAppend('div', dataDiv, {
+            class: 'contributor-data',
+          });
+          createAndAppend('div', cotributorData, {
+            class: 'contributor-login',
+            text: contributor.login,
+          });
+          createAndAppend('div', cotributorData, {
+            class: 'contributor-badge',
+            text: contributor.contributions,
+          });
+        });
+      }
+    });
+  }
   function main(url) {
     fetchJSON(url, (err, data) => {
       const root = document.getElementById('root');
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        creatHeader(data, root);
+        createDescription(data[0], mainContainerForRepo);
+        createContributors(data[0], mainContainerForRepo);
```

Comment:
Your `createContributors()`  function expects the root element as its second parameter.